src/core/core.controller.js

```diff
@@ -905,6 +906,7 @@ class Chart {
     }
 
     this.notifyPlugins('destroy');
+    this.notifyPlugins('afterDestroy');
```

Comment:
I'd move this after removing from instances. But keep the `destroy` hook as is until its removed.
Could add a comment like
// TODO v4: remove destroy hook